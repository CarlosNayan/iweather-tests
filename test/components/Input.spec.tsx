import { render, screen } from "@testing-library/react-native";
import { Input } from "@components/Input";

describe("Component: Input", () => {
  it("should render correctly", () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="placeholder" />
    );

    expect(screen.queryByTestId("loading")).toBeNull();
    expect(getByPlaceholderText("placeholder")).toBeTruthy();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("should render isLoading correctly", () => {
    render(<Input placeholder="placeholder" isLoading />);

    expect(screen.getByTestId("loading")).toBeTruthy();
  });
});
