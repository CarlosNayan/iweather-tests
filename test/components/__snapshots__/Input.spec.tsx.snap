// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Input should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#22222F",
      "borderRadius": 8,
      "flexDirection": "row",
      "fontFamily": "Nunito_400Regular",
      "fontSize": 16,
      "height": 56,
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
>
  <TextInput
    placeholder="placeholder"
    placeholderTextColor="#7F7F98"
    style={
      {
        "color": "#FFFFFF",
        "flex": 1,
      }
    }
  />
</View>
`;
